sycl/test-e2e/Annotated_arg_ptr/annotated_arg.cpp

```diff
@@ -31,6 +31,19 @@ int main() {
   for (int i = 0; i < 4; i++)
     d_ptr[i] = i;
 
+  auto *e = malloc_shared<int>(4, Q);
```

Comment:
Testing with ints is not sufficient. Create some templated struct like:
```c++
template<typename T>
struct MyStruct{
  T data;
  // ... arithmetic operators
}
```
Add tests to check these operations for non pointer types. I think header changes might be needed if you have only added the operators for the `annotated_arg<T*>` specialization. Specifically check these:
```c++

annotated_arg<MyStruct, ...> a;
annotated_arg<MyStruct, ...> b;
MyStruct c;
q.single_task([=]{
  ... = a + b;
  ... = b + c;

}).wait();
```